go/worker/storage/worker.go

```diff
@@ -70,7 +70,9 @@ func (w *Worker) registerRuntime(commonNode *committeeCommon.Node) error {
 		"runtime_id", id,
 	)
 
-	rp, err := w.registration.NewRuntimeRoleProvider(node.RoleComputeWorker, id)
+	// Since the storage node is always coupled with another role, make sure to not add any
+	// particular role here -- instead only serve as a registration gatekeeper.
+	rp, err := w.registration.NewRuntimeRoleProvider(node.RoleEmpty, id)
```

Comment:
Does this return an error as the empty role is not a single role? See [newRoleProvider](https://github.com/oasisprotocol/oasis-core/blob/master/go/worker/registration/worker.go#L772).

Reply:
It shouldn't since I also updated the method?